pages/cas/clubs/[id]/edit.vue

```diff
@@ -0,0 +1,181 @@
+<script setup lang="ts">
+import { useRoute } from 'vue-router'
+import { format } from 'date-fns'
+import { useForm } from 'vee-validate'
+import { toTypedSchema } from '@vee-validate/zod'
+import * as z from 'zod'
+import Calendar from '../../../../components/ui/calendar/Calendar.vue'
+import { Card, CardContent, CardDescription, CardHeader, CardTitle } from '~/components/ui/card'
+import { Button } from '~/components/ui/button'
+import { cn } from '~/lib/utils'
+import { Textarea } from '~/components/ui/textarea'
+import { useToast } from '~/components/ui/toast/use-toast'
+import { Toaster } from '~/components/ui/toast'
+import type { Club, Clubs } from '~/types/clubs'
+
+const emit = defineEmits(['refresh'])
+const { data } = await useFetch<Clubs>('/api/club/all_details')
+const clubs = data.value!
+const route = useRoute()
+const id = route.params.id // Fetch current Club ID via route params
+
+// Filter clubs based on C_GroupsID and include information at the same level as groups
+const filteredClubs = Object.values(clubs).flatMap(clubCategory =>
+  clubCategory.filter((club: Club) =>
+    club.groups.some(group => group.C_GroupsID === id),
+  ).map((club: Club) => ({
+    ...club, // Spread to include all same-level information
+    groups: club.groups.filter(group => group.C_GroupsID === id), // Filter groups to only include those that match the ID
+  })),
+) as Club[]
+
```

Comment:
Don't. Fetch from `/api/user/all_clubs` then check if this user has this club with the club id and if this guy is the president of that club.